unrouted_handler.go

```diff
@@ -460,7 +488,7 @@ func (handler *UnroutedHandler) writeChunk(id string, info FileInfo, w http.Resp
 	offset := info.Offset
 
 	// Test if this upload fits into the file's size
-	if offset+length > info.Size {
+	if !info.SizeIsDeferred && offset+length > info.Size {
 		return ErrSizeExceeded
 	}
 
```

Comment:
The following lines where maxSize is defined also need a consideration if the size is deferred. Something like:

```go
maxSize := info.Size - offset
// If the upload's length is deferred and the PATCH request does not contain the Content-Length
// header (which is allowed if 'Transfer-Encoding: chunked' is used), we still need to set limits for
// the body size.
if info.SizeIsDeferred {
	if handler.config.MaxSize > 0 {
		// Ensure that the upload does not exceed the maximum upload size
		maxSize = handler.config.MaxSize - offset
	} else {
		// If no upload limit is given, we will allowed arbitrary sizes
		maxSize = math.MaxInt64
	}
}
if length > 0 {
	maxSize = length
}
```